components/gitpod-db/src/typeorm/migration/1663784254956-IndexPhoneNumber.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { columnExists } from "./helper/helper";
+
+const D_B_USER = "d_b_user";
+const COL_PHONE_NUMBER = "verificationPhoneNumber";
+
+export class IndexPhoneNumber1663784254956 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        if (!(await columnExists(queryRunner, D_B_USER, COL_PHONE_NUMBER))) {
+            await queryRunner.query(
+                `ALTER TABLE ${D_B_USER} ADD INDEX (${COL_PHONE_NUMBER}), ALGORITHM=INPLACE, LOCK=NONE `,
```

Comment:
What's the expected shape of the phone number here? Is it including spaces? Is it including country codes? Are the country codes encoded as + or 00? Would any of these formats cause issues when checking for the number of times a number has been used?

Reply:
There are a couple of allowed shapes, which could be exploited to reuse the same number. Is that what you are asking?

Reply:
Basically. Could we in the least do the following transformations:

Reply:
We could store the number in https://en.wikipedia.org/wiki/E.164. I'll check later if that's easy enough.